### PR TITLE
Make it possible to use `Registry` without requiring `ArbiterService`

### PR DESCRIPTION
**Pull request by frewsxcv**
## PR Type

Docs

## PR Checklist
Check your PR fulfills the following:

- [x] ~Tests for the changes have been added / updated.~
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

One of the downsides of the `ArbiterService` trait is that it requires `Default`, which isn't always easy to implement. This pull request makes it possible to use `Registry` without requiring `ArbiterService`, by removing the `ArbiterService`, and introducing a `try_get` method.

### CODE REVIEW

**Comment by frewsxcv** [comment]
@robjtede Whenever you have a few minutes, do you have any feedback about this one?

**Comment by robjtede** [comment]
FYI this is waiting on the other two checklist items before I can merge it.

**Review by robjtede — APPROVED** [review]
LGTM 👍🏻

**Review by robjtede — APPROVED** [review]

**Pull request merged**
